streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -47,7 +47,8 @@ private[streaming] class FileBasedWriteAheadLog(
     logDirectory: String,
     hadoopConf: Configuration,
     rollingIntervalSecs: Int,
-    maxFailures: Int
+    maxFailures: Int,
+    closeAfterWrite: Boolean = false
```

Comment:
I didn't want to break existing code. Apparently `FileBasedWriteAheadLog` is instantiated in only 4 spots, therefore I can manually add the `closeAfterWrite = false` parameter to all of them. It could be okay to keep it though as the method is private[streaming] and we don't need to keep Java compatibility as it is not exposed